tasm-test/src/LayoutBenchmarkTest.cc

```diff
@@ -0,0 +1,198 @@
+    runBenchmark(videoName, entry, iteration, outputFile, -1, "tileLayoutForId", [id] (TiledVideoManager &manager){
+        manager.tileLayoutForId(id);
+    });
+}
+
+void testLocationOfTileForId (std::string videoName, std::shared_ptr<TiledEntry> entry, int frames, int id, int iteration, std::ofstream &outputFile) {
+    for (int frameNumber = 0; frameNumber < frames; frameNumber += frames / 4) {
+        runBenchmark(videoName, entry, iteration, outputFile, frameNumber, "locationOfTileForId", [frameNumber, id] (TiledVideoManager &manager){
+            manager.locationOfTileForId(frameNumber, id);
+        });
+    }
+}
+
+void testTotalWidth (std::string videoName, std::shared_ptr<TiledEntry> entry, int iteration, std::ofstream &outputFile) {
+    runBenchmark(videoName, entry, iteration, outputFile, -1, "testTotalWidth", [] (TiledVideoManager &manager){
+        manager.totalWidth();
+    });
+}
+
+void testTotalHeight (std::string videoName, std::shared_ptr<TiledEntry> entry, int iteration, std::ofstream &outputFile) {
+    runBenchmark(videoName, entry, iteration, outputFile, -1, "totalHeight", [] (TiledVideoManager &manager){
+        manager.totalHeight();
+    });
+}
+
+void testLargestWidth (std::string videoName, std::shared_ptr<TiledEntry> entry, int iteration, std::ofstream &outputFile) {
+    runBenchmark(videoName, entry, iteration, outputFile, -1, "largestWidth", [] (TiledVideoManager &manager){
+        manager.largestWidth();
+    });
+}
+
+void testLargestHeight (std::string videoName, std::shared_ptr<TiledEntry> entry, int iteration, std::ofstream &outputFile) {
+    runBenchmark(videoName, entry, iteration, outputFile, -1, "largestHeight", [] (TiledVideoManager &manager){
+        manager.largestHeight();
+    });
+}
+
+void testMaximumFrame (std::string videoName, std::shared_ptr<TiledEntry> entry, int iteration, std::ofstream &outputFile) {
+    runBenchmark(videoName, entry, iteration, outputFile, -1, "maximumFrame", [] (TiledVideoManager &manager){
+        manager.maximumFrame();
+    });
+}
+
+class LayoutBenchmarkTestFixture : public ::testing::Test {
+public:
+    LayoutBenchmarkTestFixture() {
+    }
+};
+
+TEST_F(LayoutBenchmarkTestFixture, runBenchmarks) {
+    // set test configuration
+    std::unordered_map<std::string, std::string> options {
+            {EnvironmentConfiguration::DefaultLayoutsDB, "benchmark-layout.db"},
+            {EnvironmentConfiguration::CatalogPath, "benchmark-resources"},
+            {EnvironmentConfiguration::DefaultLabelsDB, "benchmark-labels.db"}
+    };
+    auto config = EnvironmentConfiguration::instance(EnvironmentConfiguration(options));
+
+    // initialize video info
+    std::string labelPaths[NUM_VIDEOS] LABEL_PATHS;
+    std::string videoPaths[NUM_VIDEOS] VIDEO_PATHS;
+    std::string labels[NUM_VIDEOS] VIDEO_LABELS;
+    std::string names[NUM_VIDEOS] VIDEO_NAMES;
+    int numFrames[NUM_VIDEOS] NUM_FRAMES;
+    int ids[NUM_VIDEOS] IDS;
```

Comment:
Alternatively, you could define all of these constants as their actual types at the top of the file and just use them here.